cl_manycore/libraries/bsg_manycore_cuda.cpp

```diff
@@ -458,6 +472,49 @@ int hb_mc_tile_group_enqueue (	hb_mc_device_t* device,
 
 
 
+/**
+ * Takes in a kernel name, argument count and list of arguments 
+ * and initializes a hb_mc_kernel_t struct for the tile group.
+ * @param[in]  tg            Pointer to tile group. 
+ * @param[in]  name          Kernel name that is to be executed on tile group
+ * @param[in]  argc          Number of input arguments to the kernel
+ * @param[in]  argv          List of input arguments to the kernel
+ * @return HB_MC_SUCCESS on success and HB_MC_FAIL on failure. 
+ */
+__attribute__((warn_unused_result))
+static int hb_mc_tile_group_kernel_init (	hb_mc_tile_group_t *tg, 
+						char* name, 
+						uint32_t argc, 
+						uint32_t argv[]) {
+	int error;
+	tg->kernel = (hb_mc_kernel_t *) malloc (sizeof(hb_mc_kernel_t));
+	if (tg->kernel == NULL) { 
+		bsg_pr_err("%s: failed to allocated space for hb_mc_kernel_t struct.\n", __func__);
+		return HB_MC_NOMEM;
+	}
+	tg->kernel->name = strdup (name); 
+	if (tg->kernel->name == NULL) { 
+		bsg_pr_err("%s: failed to allocate space on device for tile group's kernel's name.\n", __func__);
+		return HB_MC_NOMEM;
+	}
+	tg->kernel->argc = argc;
+	tg->kernel->argv = (uint32_t *) malloc (tg->kernel->argc * sizeof(uint32_t)); 
+	if (tg->kernel->argv == NULL) { 
+		bsg_pr_err("%s: failed to allocate space on devcie for kernel's argument list.\n", __func__); 
+		return HB_MC_NOMEM;
+	}
+	memcpy (tg->kernel->argv, argv, argc * sizeof(uint32_t) / sizeof(uint8_t)); 	
```

Comment:
I wouldn't divide by `sizeof(uint8_t)` -- this code in unlikely to ever run on a machine where `sizeof(uint8_t)!=1`

Reply:
Wait, what?

Reply:
@drichmond you'd be amazed the crazy stuff people come up with.

Reply:
Hmmm, so should I just go with argc * 4 instead?

Reply:
No, that is fine. We're saying do this:
```c
argc * sizeof(uint32_t)
```
not this:
```c
argc * sizeof(uint32_t) / sizeof(uint8_t)
```

Reply:
Ahh I see, makes sense, will do, Thanks!